index.js

```diff
@@ -30,27 +30,19 @@ module.exports = (stream, opts) => {
 	}
 
 	const completion = new Promise((resolve, reject) => {
-		function onEnd(result) {
-			if (awaited) {
-				awaited.then(resolve);
+		if (awaited) {
```

Comment:
Any reason you refactored the order?

Is it correct that the only actual addition is the `reject` in the `awaited.then(resolve, reject)`?